lib/rules/no-missing-g-flag.ts

```diff
@@ -0,0 +1,138 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+import type { ExpressionReference } from "../utils/ast-utils"
+import {
+    extractExpressionReferences,
+    isKnownMethodCall,
+} from "../utils/ast-utils"
+import { createTypeTracker } from "../utils/type-tracker"
+
+/**
+ * Parse option
+ */
+function parseOption(
+    userOption:
+        | {
+              strictTypes?: boolean
+          }
+        | undefined,
+) {
+    let strictTypes = true
+    if (userOption) {
+        if (userOption.strictTypes != null) {
+            strictTypes = userOption.strictTypes
+        }
+    }
+
+    return {
+        strictTypes,
+    }
+}
+
+export default createRule("no-missing-g-flag", {
+    meta: {
+        docs: {
+            description:
+                "disallow missing `g` flag in patterns used in `String#matchAll` and `String#replaceAll`",
+            category: "Possible Errors",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    strictTypes: { type: "boolean" },
+                },
+                additionalProperties: false,
+            },
+        ],
+        messages: {
+            missingGlobalFlag:
+                "The pattern given to the argument of `String#{{method}}()` requires the `g` flag, but is missing it.",
+        },
+        type: "problem",
+    },
+    create(context) {
+        const { strictTypes } = parseOption(context.options[0])
+        const typeTracer = createTypeTracker(context)
+
+        /**
+         * Create visitor
+         */
+        function createVisitor(
+            regexpContext: RegExpContext,
+        ): RegExpVisitor.Handlers {
+            const { regexpNode, flags, ownsFlags } = regexpContext
+
+            if (flags.global || !ownsFlags) {
```

Comment:
Should we really require `ownsFlags`? If a pattern doesn't own its flags, it will still cause a runtime error. The only difference is that we can't provide a fix (and `fixReplaceFlags` will automatically handle this).

Reply:
Thank you for checking this out.
I should have used `flagsString == null` instead.